libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -418,50 +428,23 @@ void AP_GPS::detect_instance(uint8_t instance)
     switch (_type[instance]) {
     // user has to explicitly set the MAV type, do not use AUTO
     // do not try to detect the MAV type, assume it's there
-    case GPS_TYPE_MAV: {
+    case GPS_TYPE_MAV:
         dstate->auto_detected_baud = false; // specified, not detected
         new_gps = new AP_GPS_MAV(*this, state[instance], nullptr);
         goto found_gps;
         break;
+    default:
+        break;
     }
 
-    // user has to explicitly set the UAVCAN type, do not use AUTO
-    case GPS_TYPE_UAVCAN: {
 #if HAL_WITH_UAVCAN
-        dstate->auto_detected_baud = false; // specified, not detected
-
-        uint8_t can_num_drivers = AP::can().get_num_drivers();
-
-        for (uint8_t i = 0; i < can_num_drivers; i++) {
-            AP_UAVCAN *ap_uavcan = AP_UAVCAN::get_uavcan(i);
-            if (ap_uavcan == nullptr) {
-                continue;
-            }
-
-            uint8_t gps_node = ap_uavcan->find_gps_without_listener();
-            if (gps_node == UINT8_MAX) {
-                continue;
-            }
-
-            new_gps = new AP_GPS_UAVCAN(*this, state[instance], nullptr);
-            ((AP_GPS_UAVCAN*) new_gps)->set_uavcan_manager(i);
-            if (ap_uavcan->register_gps_listener_to_node(new_gps, gps_node)) {
-                if (AP::can().get_debug_level_driver(i) >= 2) {
-                    printf("AP_GPS_UAVCAN registered\n\r");
-                }
-                goto found_gps;
-            } else {
-                delete new_gps;
-            }
+    if (new_gps == nullptr && _type[instance] == GPS_TYPE_UAVCAN) {
```

Comment:
Why it was moved out of switch if `_type[instance] == GPS_TYPE_UAVCAN` was still left as a condition?
Also as per code - should not new_gps always be nullptr here?

Reply:
Yes, should be moved to switch case.